Handle list-based select criteria

mfb2 opened the pull request:
Prior to this commit, use of the `:select` query criteria resulted in a
failure of the `ecto_ldap` adapter to return results due to a mismatched
function head expectation.

This commit resolves the bug by conforming the expected data to use
flattened lists instead of lists-of-lists, as well as consistent pruning
of the selected attributes to be returned from the adapter.

Tests which exercise the expected response have been added to ensure
proper functioning going forward.

Commit:
Handle list-based select criteria

Prior to this commit, use of the `:select` query criteria resulted in a
failure of the `ecto_ldap` adapter to return results due to a mismatched
function head expectation.

This commit resolves the bug by conforming the expected data to use
flattened lists instead of lists-of-lists, as well as consistent pruning
of the selected attributes to be returned from the adapter.

Tests which exercise the expected response have been added to ensure
proper functioning going forward.

OvermindDL1 commented:
You know, as it looks like this library 'gets' data from ldap only without inserting, have you thought of using a library like my [ldap_ex](https://hex.pm/packages/ldap_ex) library?  It reimplements Erlang's ldap module using binaries with Elixir wrappers.  I have the full LDAP spec compiled in so I have been planning to add write capabilities to it sometime, just not got around to it yet, but it is much easier to use and can handle its own connections in the config if you so wish (or you could).

Commit:
Handle bound variable on left-hand side of where clause

Prior to this commit, variable binding was only handled on the
right-hand side of `:where` statements.  This commit introduces the
ability to bind variables on the left-hand side of `:where` clauses when
conducting `ecto_ldap` queries.
